front/src/store/tv/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,65 @@
+import { tvReducer } from '../reducers';
+import {
+  REQUEST_BASIC_SHOW_INFO_SUCCEEDED,
+  SAVE_EPISODE_DATA,
+  SAVE_SEARCH_QUERY,
+  SET_CALENDAR_EPISODES,
+} from '../actions';
+
+describe('tv reducer', () => {
+  const commonState = {
+    savedQueries: [{ query: '1' }],
+    episodeData: {},
+    basicShowInfo: {},
+    calendarEpisodesForDisplay: [],
+  };
+
+  it('saves query', () => {
+    const action = { type: SAVE_SEARCH_QUERY, payload: { query: '1' } };
+    expect(tvReducer(undefined, action)).toEqual(commonState);
+  });
+
+  it('does not add same query', () => {
+    const initialState = {
+      savedQueries: [{ query: '1' }] as any,
+      episodeData: {},
+      basicShowInfo: {},
+      calendarEpisodesForDisplay: [],
+    };
+    const action = { type: SAVE_SEARCH_QUERY, payload: { query: '1' } };
+    expect(tvReducer(initialState, action)).toEqual(commonState);
+  });
+
+  it('saves episode data', () => {
+    const expectedState = {
+      savedQueries: [],
+      episodeData: { episode: 'test' },
+      basicShowInfo: {},
+      calendarEpisodesForDisplay: [],
+    };
+    const action = { type: SAVE_EPISODE_DATA, payload: { episode: 'test' } };
+    expect(tvReducer(undefined, action)).toEqual(expectedState);
```

Comment:
should this `undefined` be `initialState` instead?

Reply:
It can be but we don't need to define it except in the 'does not add same query' test

Reply:
Ok, keeping as you have it looks good